### DISCUSSION TITLE
Is it possible to enable xcomarg more external processing method?

### DISCUSSION THREAD

**Discussion by appassionate** [apache/airflow#22915] [Ideas] [answered]
Hi, I want to use taskAPI for conveniently parameter passing.
I expected the "print_value" print "a" but it will print "None". I guess XComArg pull xcom value only when a operator is being executed. But it might not affect the predefinition for the method I get from the XCom. Or I have missed some usage of XComArg? 

I imaged that I can do in an intuitive style.
```python
param = params.resolve(load_func= lambda x: x[1]) #params is a xcomarg while param might be a derived xcomarg?
```

otherwise, I have to:
```python
@task
def get_detailed_value(params, keys):
    return params[keys]
param = get_detailed_value(params, 1)
```

```python
from airflow.models import DAG
from airflow.operators.python import PythonOperator, task
from airflow.utils.dates import days_ago
from airflow.utils.edgemodifier import Label
from airflow import DAG

default_args={}

@task
def gen_params():

    return {1:'a',2:'b'}

@task
def print_value(t_value):

    print("value: ",t_value)


with DAG('test',
        start_date=days_ago(2),
        max_active_runs=3,
        schedule_interval=None,
        default_args=default_args,
        catchup=False
    ) as dag:

    params = gen_params()
    param = params[1]
    test = print_value(param)
```

Regards! : )

**Comment by potiuk** [comment, marked as answer]
It does not work this way. You cannot pas parameters this way to decorated function. you can pass other tasks and then they will produce outputs that will be passed to the task see https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html it explains it on examples.
